module/programs/helper.js

```diff
@@ -150,6 +197,827 @@ module.exports = class ProgramsHelper {
+
+      if (Array.isArray(roles) && roles.length > 0) {
+        let userRoles = await userRolesHelper.roleDocuments(
+          {
+            code: { $in: roles },
+          },
+          ["_id", "code"]
+        );
+
+        if (!(userRoles.length > 0)) {
+          return resolve({
+            status: httpStatusCode.bad_request.status,
+            message: messageConstants.apiResponses.INVALID_ROLE_CODE,
+          });
+        }
+
+        await programsQueries
+          .findOneAndUpdate(
+            {
+              _id: programId,
+            },
+            {
+              $pull: { "scope.roles": { code: messageConstants.common.ALL_ROLES } },
+            },
+            { new: true }
+          );
+
+        updateQuery["$addToSet"] = {
+          "scope.roles": { $each: userRoles },
+        };
+      } else {
+        if (roles === messageConstants.common.ALL_ROLES) {
+          updateQuery["$set"] = {
+            "scope.roles": [{ code: messageConstants.common.ALL_ROLES }],
+          };
+        }
+      }
+
+      let updateProgram = await programsQueries
+
+        .findOneAndUpdate(
+          {
+            _id: programId,
+          },
+          updateQuery,
+          { new: true }
+        )
+;
+
+      if (!updateProgram || !updateProgram._id) {
+        throw {
+          message: messageConstants.apiResponses.PROGRAM_NOT_UPDATED,
+        };
+      }
+
+      return resolve({
+        message: messageConstants.apiResponses.ROLES_ADDED_IN_PROGRAM,
+        success: true,
+      });
+    } catch (error) {
+      return resolve({
+        success: false,
+        status: error.status
+          ? error.status
+          : httpStatusCode["internal_server_error"].status,
+        message: error.message,
+      });
+    }
+  });
+}
+
+
+  /**
+   * Add entities in program.
+   * @method
+   * @name addEntitiesInScope
+   * @param {String} programId - Program Id.
+   * @param {Array} entities - entities data.
+   * @returns {JSON} - Added entities data.
+   */
+
+  static addEntitiesInScope(programId, entities,userToken) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let programData = await programsQueries.programDocuments(
+          {
+            _id: programId,
+            scope: { $exists: true },
+            isAPrivateProgram: false,
+          },
+          ["_id", "scope.entityType"]
+        );
+
+        if (!(programData.length > 0)) {
+          throw {
+            message: messageConstants.apiResponses.PROGRAM_NOT_FOUND,
+          };
+        }
+
+        let entityIds = [];
+        let bodyData = {};
+        let locationData = gen.utils.filterLocationIdandCode(entities);
+
+        if (locationData.ids.length > 0) {
+          bodyData = {
+            "registryDetails.code": {$in:locationData.ids},
+            entityType: programData[0].scope.entityType,
+          };
+          let entityData = await entityManagementService.locationSearch(bodyData,userToken);
```

Comment:
@praveenKDass function name should be changing . hope this will be covered during entity-management service integration

Reply:
Covering this in entity-management PR review will keep as like that for now @VISHNUDAS-tunerlabs